python/ray/air/util/tensor_extensions/arrow.py

```diff
@@ -720,7 +721,9 @@ def _to_numpy(self, index: Optional[int] = None, zero_copy_only: bool = False):
             arrs = [self._to_numpy(i, zero_copy_only) for i in range(len(self))]
             # Return ragged NumPy ndarray in the ndarray of ndarray pointers
             # representation.
```

Comment:
Nit: Could you extend the note to explain why `np.empty` is necessary? Don't think it'd be obvious to readers.